MIDIchlorians/MIDIchlorians/CloudManager.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  CloudManager.swift
+//  MIDIchlorians
+//
+//  Created by Advay Pal on 1/4/17.
+//  Copyright © 2017 nus.cs3217.a0118897. All rights reserved.
+//
+
+import SwiftyDropbox
+
+class CloudManager {
+
+    static let instance = CloudManager()
+    private let dataManager = DataManager.instance
+    private var client: DropboxClient? {
+        return DropboxClientsManager.authorizedClient
+    }
+
+    private init() {
+        //Just to ensure initialiser is private
+    }
+
+    func loadFromDropbox() {
+        loadAudios()
+        loadAnimations()
+        loadSessions()
+    }
+
+    private func loadAudios() {
+        guard let docsURL = FileManager.default.urls(for: .documentDirectory,
+                                                     in: .userDomainMask).last else {
+           return
+        }
+        client?.files.listFolder(path: "/\(Config.AudioFolderName)/").response { response, _ in
+            guard let result = response else {
+                return
+            }
+            for sample in result.entries {
+                guard self.isSoundFile(sample.name) else {
+                    continue
+                }
+                let filePath = "/\(Config.AudioFolderName)/\(sample.name)"
+                let url = docsURL.appendingPathComponent(sample.name)
+                let destination: (URL, HTTPURLResponse) -> URL = { temporaryURL, response in
+                    return url
+                }
+                self.client?.files.download(path: filePath, overwrite: true, destination: destination).response { response, error in
+                    print(response)
+
+                }
+                _ = self.dataManager.saveAudio(sample.name)
```

Comment:
Yeah I think we should move this to the closure. Another issue is that once all downloads are complete we'll need to let the frontend know, so that the list of all sessions/audios/animations can be reloaded